crates/hir_ty/src/diagnostics/decl_check.rs

```diff
@@ -181,8 +181,12 @@ impl<'a> DeclValidator<'a> {
                 AttrDefId::ExternBlockId(id) => Some(id.lookup(self.db.upcast()).container.into()),
                 // These warnings should not explore macro definitions at all
                 AttrDefId::MacroDefId(_) => None,
-                // Will never occur under an enum/struct/union/type alias
```

Comment:
I think this pr should be the fix but have no idea why origin comment says this.

Reply:
Ye that comment seems to be wrong